src/test/java/steps/YoutubeHTML5VideoStepDefinitions.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    @When("^user presses the play button$")
+    public void the_user_presses_the_play_button() throws Throwable {
+        js.executeScript(videoElement + ".play();"); //Note that this is directly using the API to play, not pressing the play button
+    }
+
+    @Then("^validate the video is paused$")
+    public void validate_the_video_pauses() throws Throwable {
+        assertTrue((Boolean) js.executeScript("return " + videoElement + ".paused;"));
+    }
+
+    @When("^user waits for \"([^\"]*)\" seconds$")
+    public void the_user_waits_for_seconds(String waitTime) throws Throwable {
+        new WebDriverWait(driver, Long.valueOf(waitTime));
+    }
+
+    @When("^user presses the size button$")
+    public void userPressesTheSizeButton() throws Throwable {
+        boolean goingIntoTheaterMode = driver.findElements(By.className("watch-stage-mode")).size() == 0;
+
+        driver.findElement(By.className("ytp-size-button")).click();
+
+        // We have to wait for theater mode transition because it takes some time
+        if (goingIntoTheaterMode) {
+            new WebDriverWait(driver, 5).until(
+                    ExpectedConditions.attributeContains(
+                            driver.findElement(By.id("page")),
+                            "class",
+                            "watch-stage-mode"));
+        } else {    // This one inverts the assumption with ExpectedConditions.not()
+            new WebDriverWait(driver, 5).until(
+                    ExpectedConditions.not(ExpectedConditions.attributeContains(
+                            driver.findElement(By.id("page")),
+                            "class",
+                            "watch-stage-mode")));
+        }
+        // new WebDriverWait(driver, 5).until(ExpectedConditions.elementToBeClickable(By.className("ytp-size-button")));
+    }
+
+    @When("^user presses the fullscreen button$")
+    public void user_presses_the_fullscreen_button() throws Throwable {
+        boolean goingIntoFullscreen = driver.findElements(By.className("ytp-fullscreen")).size() == 0;
+
+        driver.findElement(By.className("ytp-fullscreen-button")).click();
+
+        // We have to Wait for fullscreen because it takes some time
+        if (goingIntoFullscreen) {
+            new WebDriverWait(driver, 5).until(
+                    ExpectedConditions.attributeContains(
+                            driver.findElement(By.cssSelector(".html5-video-player")),
+                            "class",
+                            "ytp-fullscreen"));
+        } else {    // This one inverts the assumption with ExpectedConditions.not()
+            new WebDriverWait(driver, 5).until(
+                    ExpectedConditions.not(ExpectedConditions.attributeContains(
+                            driver.findElement(By.cssSelector(".html5-video-player")),
+                            "class",
```

Comment:
WebDriverWaits all use an object inside the api called a Sleeper which use thread sleeps. There are no such thing as true implicit waits inside selenium. We made a simple time loop that can be used without risk of thread lock.